lib/args.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const chalk = require('chalk');
+const utils = require('./utils');
+
+function Args() {
+  this.details = {
+    options: [],
+    commands: [],
+    examples: []
+  };
+
+  // Configuration defaults
+  this.config = {
+    help: true,
+    version: true,
+    usageFilter: null,
+    value: null,
+    name: null,
+    mainColor: 'yellow',
+    subColor: 'dim'
+  };
+
+  this.printMainColor = chalk;
+  this.printSubColor = chalk;
+}
+
+// Internal helpers
+Args.prototype.handleType = utils.handleType;
```

Comment:
Also can we put all of these (same for the public ones below) into an object for each and then iterate over that object and assign them to the prototype inside that loop? Because right now, we're repeating a lot of code... 😊